From: yui-knk <spiketeika@gmail.com>
Date: Thu, 18 Jan 2024 23:41:04 +0900
Subject: Use index for referring to symbols in `args` rule instead of named
 references

In `args: args ',' arg_splat`, `args` is not unique name.
Currently the associated rule is interpreted as
`$$ = rest_arg_append(p, $$, $3, &@$);`.
The action works as expected because `$$` is initialized with
`$1` before each action is executed.
However it's misleading then change to use index.
---
 parse.c | 4 ++--
 parse.y | 4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/parse.c b/parse.c
index 46e0b63..e408042 100644
--- a/parse.c
+++ b/parse.c
@@ -12747,9 +12747,9 @@ yyreduce:
 #line 3801 "parse.y"
                   {
                     /*%%%*/
-                        (yyval.node) = rest_arg_append(p, (yyval.node), (yyvsp[0].node), &(yyloc));
+                        (yyval.node) = rest_arg_append(p, (yyvsp[-2].node), (yyvsp[0].node), &(yyloc));
                     /*% %*/
-                    /*% ripper: args_add_star!($args, $arg_splat) %*/
+                    /*% ripper: args_add_star!($1, $3) %*/
                     }
 #line 12755 "parse.c"
     break;
diff --git a/parse.y b/parse.y
index f8e21dc..55a304b 100644
--- a/parse.y
+++ b/parse.y
@@ -3800,9 +3800,9 @@ args		: arg_value
                 | args ',' arg_splat
                     {
                     /*%%%*/
-                        $$ = rest_arg_append(p, $args, $arg_splat, &@$);
+                        $$ = rest_arg_append(p, $1, $3, &@$);
                     /*% %*/
-                    /*% ripper: args_add_star!($args, $arg_splat) %*/
+                    /*% ripper: args_add_star!($1, $3) %*/
                     }
                 ;
 
